[Workflows] Display workflow execution overview with trigger and context

talboren opened the pull request:
## Summary

This PR implements the execution overview panel for workflow executions, displaying trigger information and execution context.

### Changes:

**Overview Step**
- Added Overview pseudo-step that displays at the top of the step execution tree
- Auto-selects Overview when opening an execution
- Created  helper function

**UI Components**
- Implemented  component with:
  - Execution status with icon and relative time ("just now")
  - Test run badge indicator
  - Total execution duration with clock icon
  - Execution started and ended timestamps with millisecond precision
  - Execution context data table with copyable field paths
- Refactored to use EUI Flex components throughout for better consistency
- Added proper spacing and styling matching Figma designs

**Trigger Detection**
- Fixed manual trigger detection to show trigger even when inputs are empty
- Changed logic from checking if inputs has keys to checking if inputs property exists

**Tree View Styling**
- Reduced expansion arrow size from 16px to 12px for better visual hierarchy

**Field Path Actions**
- Made execution context fields copyable with appropriate field path notation
- Overview fields use empty prefix (e.g., `execution.id`, `workflow.name`)

### Related Issue
Relates to elastic/security-team#14698

### Testing
- Tested with manual executions (with and without inputs)
- Tested with scheduled executions
- Verified test run badge appears correctly
- Verified execution times display with millisecond precision
- Verified field paths are copyable

### Screenshots

<img width="3024" height="1646" alt="CleanShot 2025-11-30 at 21 42 07@2x" src="https://github.com/user-attachments/assets/d0be1cef-c589-42d9-8d39-d6fd1ba694d0" />
<img width="3024" height="1646" alt="CleanShot 2025-11-30 at 21 41 32@2x" src="https://github.com/user-attachments/assets/9287f50a-d74b-4661-b6c8-0b5ca225b93e" />

talboren committed:
feat: wip

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
feat: wip

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Allow multiple workflow system actions

talboren committed:
Merge branch 'multiple-workflow-system-action' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Revert "Allow multiple workflow system actions"

This reverts commit 1b5f95760bd444ea77f1f8100d49e599d33f4975.

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
wip

talboren committed:
wip

talboren committed:
wip

talboren committed:
wip

talboren committed:
wip

talboren committed:
wip

talboren commented on src/platform/plugins/shared/workflows_management/public/features/workflow_execution_detail/ui/workflow_step_execution_tree.tsx:
```diff
@@ -66,9 +70,27 @@ function convertTreeToEuiTreeViewItems(
+              display: inline-block;
+              width: 16px;
+              height: 16px;
+              background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path d="M7.04066265,13.2735746 C6.91555227,13.5197493 7.01369433,13.8207354 7.25986907,13.9458457 C7.50604381,14.0709561 7.80702987,13.9728141 7.93214025,13.7266393 L10.9463954,7.79560891 C11.1154447,7.46297732 10.8737805,7.06907652 10.5006566,7.06907652 ...
```
didn't manage to inject it to eui tree view without it

talboren committed:
Remove duplicate workflow_trigger_context.ts file

The code was duplicated in workflow_pseudo_step_context.ts. All imports
are using workflow_pseudo_step_context.ts, so removing the duplicate.

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
fix: update tests for workflow execution overview and pseudo-step handling

- Modified buildStepExecutionsTree to only add Overview pseudo-step when executionStatus is provided
- Fixed test expectations to match new behavior where inputs pseudo-step is added even for empty inputs objects
- Removed obsolete WorkflowExecutionListItem tests from workflow_execution_panel.test.tsx
- Updated workflow_step_execution_tree.test.tsx to expect 3 arguments in buildStepExecutionsTree calls

talboren committed:
clock icon

talboren committed:
fix

talboren committed:
fix

talboren committed:
feat: add comprehensive test coverage for WorkflowExecutionOverview component

- Created workflow_execution_overview.test.tsx with 24 tests
- Tests cover rendering, status display, test run badge, duration display, date formatting, edge cases, and i18n labels
- Ensured all tests pass with correct status labels (Success for COMPLETED, Error for FAILED)
- Fixed trigger type detection logic to prioritize event over inputs (scheduled/alert triggers now show correctly)

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

talboren committed:
fix: resolve type errors and import ordering in workflow_execution_overview.test.tsx

- Fixed import order to comply with eslint rules
- Changed Date objects to ISO strings to match JsonValue type
- Changed undefined values in context to empty object
- All type checks and tests passing

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Merge branch 'elastic:14698-design-task-display-workflow-trigger-and-input-context-in-execution-view' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
typo

talboren committed:
Merge branch 'main' of https://github.com/elastic/kibana into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Merge branch 'elastic:14698-design-task-display-workflow-trigger-and-input-context-in-execution-view' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

h88 reviewed (approved):
Left some minor comments for your consideration.
Additionally, I noticed an `aria-hidden` warning from EuiDataGrid which seems to be a pre existing one (not a regression).

Great work on implementing the execution overview panel!

talboren committed:
Merge branch 'main' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
fix: address CR feedback - add executionStatus tests and fix pagination overflow

1. Added comprehensive tests for buildStepExecutionsTree with executionStatus parameter:
   - Test overview pseudo-step is added when executionStatus is provided
   - Test overview is not added when executionStatus is undefined
   - Test overview is placed at position 0 before other pseudo-steps
   - Test all execution statuses (pending, running, completed, failed, cancelled, skipped)
   - Test overview is added even with empty step executions
   - Test overview is placed before nested structures

2. Fixed pagination overflow in workflow_execution_overview.tsx:
   - Added overflow: hidden and minHeight: 0 to the container
   - Applied same fix to json_data_view.tsx to properly contain pagination

talboren committed:
Merge branch 'elastic:14698-design-task-display-workflow-trigger-and-input-context-in-execution-view' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view

talboren committed:
Merge branch 'main' into 14698-design-task-display-workflow-trigger-and-input-context-in-execution-view
